pkg/common/config/auth/auth.go

```diff
@@ -28,12 +27,16 @@ type Config struct {
 
 // IsAuthTypeOIDC makes check that current auth is TypeOIDC.
 func (c *Config) IsAuthTypeOIDC() bool {
-	return c.AuthType == TypeOIDC
+	return c.AuthOIDCClientID != "" &&
```

Comment:
this seems like it's validating the config before letting us characterize it as a type of auth -- ie, it seems like it could "be" AuthTypeOIDC even if not configured right.

Reply:
it's just kind of semantic -- previously we "declared" the `AuthType`, now it must be valid config, which seems like it might be confusing at some point. A person could attempt to configure for OIDC, but some field is missing, so the error might say  "No auth type is configured" instead of "Invalid OIDC config".

Reply:
why remove the `AuthType` field?

Reply:
I think it should attempt to use the OIDC middleware if that's intended, even if some field is missing.